ethereum/p2p/src/main/java/tech/pegasys/pantheon/ethereum/p2p/netty/NettyPeerConnection.java

```diff
@@ -156,13 +156,13 @@ public boolean isDisconnected() {
   }
 
   @Override
-  public SocketAddress getLocalAddress() {
-    return ctx.channel().localAddress();
+  public InetSocketAddress getLocalAddress() {
+    return (InetSocketAddress) ctx.channel().localAddress();
```

Comment:
Is this always an `InetStocketAddress` in the manner we use netty?  Are there other things it could be, that we might actually use?

Reply:
According to the docs on `SocketAddress`:
> This class represents a Socket Address with no protocol attachment. As an abstract class, it is meant to be subclassed with a specific, protocol dependent, implementation.

And `InetSocketAddress`:
> This class implements an IP Socket Address (IP address + port number)

I think the change makes sense.  

We were casting it all over the place assuming it was an `InetSocketAddress` anyway and I think its better to just be explicit.

Reply:
Also, `InetSocketAddress` is actually the only subclass of `SocketAddress`

Reply:
I actually think we could probably just get rid of these methods.  I'll try to do that.

Reply:
It would localize where the cast is in the unlikely case it's ever a problem.  But if we can get rid of the methods entirely that is also less code, which is generally better.